packages/net/src/provider/non_blocking.rs

```diff
@@ -0,0 +1,117 @@
+use crate::{NetProvider, RequestHandler, USER_AGENT};
+use data_url::DataUrl;
+use futures_util::{stream::FuturesUnordered, StreamExt};
+use reqwest::Client;
+use std::{sync::Arc, time::Duration};
+use thiserror::Error;
+use tokio::{
+    runtime::{Handle, Runtime},
+    sync::Mutex,
+    task::JoinHandle,
+};
+use url::Url;
+use winit::{event_loop::EventLoopProxy, window::WindowId};
+
+type TaskHandle<T> = JoinHandle<Result<T, AsyncProviderError>>;
+
+pub struct AsyncProvider<T> {
+    rt: Handle,
+    client: Client,
+    futures: Mutex<FuturesUnordered<TaskHandle<T>>>,
+}
+impl<T: Send + Sync + 'static> AsyncProvider<T> {
+    pub fn new(rt: &Runtime) -> Self {
+        Self {
+            rt: rt.handle().clone(),
+            client: Client::new(),
+            futures: Mutex::new(FuturesUnordered::new()),
+        }
+    }
+    pub fn resolve<P: From<(WindowId, T)> + Send>(
+        self: Arc<Self>,
+        event_loop_proxy: EventLoopProxy<P>,
+        window_id: WindowId,
+    ) {
+        self.rt.clone().spawn(async move {
+            let mut interval = tokio::time::interval(Duration::from_millis(100));
+
+            'thread: loop {
+                interval.tick().await;
+                while let Some(ir) = self.futures.lock().await.next().await {
+                    match ir {
+                        Ok(Ok(t)) => {
+                            let e = event_loop_proxy.send_event((window_id, t).into());
```

Comment:
The `net` crate should not depend on `winit`.

You should use a channel here (probably a tokio [`mpsc::channel`](https://docs.rs/tokio/latest/tokio/sync/mpsc/fn.channel.html)) in the API of the net crate, listen for messages from that channel in the top-level crate (currently the dioxus-blitz crate).

Alternatively you could have the net crate accept a closure to call on success and have that closure wrap the event loop proxy.

`WindowId` can be coerced into a u64.

Reply:
I'm not sure how I would use a channel with the winit runtime.